week-2/extended-project/valeria_and_friends/classes/top.rb

```diff
@@ -0,0 +1,85 @@
+require "http"
+require "json"
+
+class Top
+  attr_accessor :name_main_attr
+  attr_accessor :printable_data
+
+  BASE_URL = "https://swapi.co/api/"
+  TOP_NUMBER = 5
+  RELEVANT_DATA = "results"
+
+  def initialize (query, name_main_attr, relevant_keys)
+    @query = query
+    @name_main_attr = name_main_attr
+    @relevant_keys = relevant_keys
+    @printable_data = []
+  end
+
+  def give_me_the_top
+    json_answer = fetch_data(@query)
+    results = get_results(json_answer)
+    main_attributes = get_main_attribute(results, @name_main_attr)
+    top_ordered = get_top_of_results(main_attributes, results)
+    final_top = build_final_top(@relevant_keys, top_ordered)
+
+    @printable_data = final_top
+    final_top
+  end
+
+  def fetch_data(query)
+    @http = HTTP.get(BASE_URL + query + "/")
+    JSON.parse(@http)
+  end
```

Comment:
Be careful! Revise good the api that you use, because it sometimes come divide for pages, in this case if you revise the link https://swapi.co/api/people/, in the key "Next" appear a value that show you the rest of information is in page 2.